graphic_scene.py

```diff
@@ -6,11 +6,22 @@
 from PySide2.QtCore import *
 from gui_views import state
 from sym_object import *
-from m5_calls import load_instantiate
 import string
 import random
+import collections
 import copy
 
+"""convert a dictionary with unicode keys and values to utf-8"""
+def convert(data):
```

Comment:
Just out of curiosity... why is this needed? Why are the values "unicode" and not the correct type?

Reply:
I think the way that the instance_ports are saved have to be in unicode, but I'll double check this. I know for a fact that the dictionary when loaded was being returned in unicode form, but maybe the way that we're saving might not be the most efficient.